CONTRIBUTING.md

```diff
@@ -1,16 +1,256 @@
+we recommend that you [start by learning it](https://try.github.io/) a bit.
+
+In this section, we give the bare minimum for a better experience with our
+development workflow on GitHub.
+
+[git-scm-getting-started]: https://git-scm.com/book/en/v2/Getting-Started-Git-Basics
+
+#### Setup your local RIOT repository
+
+Before you start modifying code, you need to fork the RIOT upstream repository
+from the [RIOT main github page][riot-github].
+
+If it's your first time with git, configure your name and emails:
+
+    $ git config --global user.name = "<your name here>"
+    $ git config --global user.email = "<your email address here>"
+
+Then clone locally your fork of RIOT (replace `account name` with your actual
+login on GitHub):
+
+    $ git clone git@github.com:<account name>/RIOT.git
+
+You can keep any branch of your local repository up-to-date with the upstream
+master branch with the following commands:
+
+    $ git checkout <branch name>
+    $ git pull --rebase https://github.com/RIOT-OS/RIOT.git
+
+Use it at least before opening a PR. This will ensure the PR can be merged.
+
+[riot-github]: https://github.com/RIOT-OS/RIOT
+
+#### Work on branches
+
+Avoid opening PR from the `master` branch of your fork to the master branch of
+the RIOT upstream repository: update your master branch and start a new branch
+from it.
+
+    $ git checkout master
+    $ git pull --rebase https://github.com/RIOT-OS/RIOT.git
+    $ git checkout -b <new branch>
+    # Do your changes, commit, update with latest upstream master
+    $ git push
+
+This is important because it will avoid the maintainer to have to deal with
+his master branch yours if he wants to test your changes locally.
+
+#### Add fixup commits during review
+
+To keep the history of changes easier to track for reviewers, it is recommended
+to push your review request updates in fixup commits.
+
+Let's say your PR contains 3 commits with comments: `dir1: change 1`,
+`dir2: change 2` and `dir3: change 3`.
+
+Instead of commiting changes in dir2 in a 4th commit `dir2: change 4`, you can
+use the `--fixup` option:
+
+    $ git add dir2
+    $ git commit --fixup <dir2 commit hash>
+
+#### Squash commits after review
+
+Squashing a commit is done using the rebase command of git in interactive mode:
+
+    $ git rebase master -i
+
+You can find information on rebasing in
+[Github rebase documentation][about-git-rebase].
+
+[about-git-rebase]: https://help.github.com/articles/about-git-rebase/
+
+If you used fixup commits during the review phase, squashing commits can be
+performed in a single command:
+
+    $ git rebase -i --autosquash
+
+**Don't squash your commit until a maintainer asks you to do it.**
+
+Otherwise the history of review changes is lost and for large PRs, it
+makes it difficult for the reviewer to follow them. You may also introduce
+regression and won't be able to recover them from previous commits.
```

Comment:
I'm not a native speaker, but "You may also introduce regression[s]" sounds like an invitation to me, though the sentence is warning here I think. Maybe use "It might also happen that you introduce regressions during the review and […]"